Add Unmanaged Operators support statement

adellape opened the pull request:
https://issues.redhat.com/browse/OSDOCS-969

Preview (internal):

* Added a new _Support policy for unmanaged Operators_ subsection in [**Architecture -> Installation and update**](http://file.rdu.redhat.com/~adellape/042720/unmanaged_operators/architecture/architecture-installation.html).
* Trying to figure where it should go in **Upgrading clusters** too, for example see here in [**Updating a cluster between minor versions**](http://file.rdu.redhat.com/~adellape/042720/unmanaged_operators/updating/updating-cluster-between-minor.html). But if seems like if I add it there, it should go in all of the other topics in that guide, paired with **About the OpenShift Container Platform update service**, which currently also appears in each of those topics. **Update:** Ended up just making "Additional resources" links back to the Arch guide from each of the "Upgrading clusters" topics.

adellape commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+// * updating/updating-cluster-between-minor.adoc
+
+[id="unmanaged-operators_{context}"]
+= Unmanaged Operators
```
Was trying to get the word "support" in the title, like simply "Unmanaged Operators support", though at a glance that makes it look like it's supported. "Supportability" is awkward, as is "non-support". And "unsupport" would make it two "un-" words in a single title...

lamek replied:
Support for unmanaged operators.
Support policy for unmanaged operators.

kalexand-rh replied:
I'd be tempted to call it "Operator management states" and add a paragraph to the update topic that says that you need to care about them before you update.

It also feels like you might be able to extract a procedure from this concept, but I don't know how many people have disabled their CVO management.

adellape replied:
> It also feels like you might be able to extract a procedure from this concept, but I don't know how many people have disabled their CVO management.

Definitely could, mostly wondering about should.

kalexand-rh replied:
> > It also feels like you might be able to extract a procedure from this concept, but I don't know how many people have disabled their CVO management.
> 
> Definitely could, mostly wondering about should.

I don't have the data for that, but if it's something that you expect people to need to do instead of just data you want them to have, it should be a procedure.

adellape replied:
Yeah, that's the question. Waiting for support folks opinion. @sferich888 ?

adellape commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+* **Individual Operator configuration**
++
+Individual Operators have a `managementState` parameter in their configuration,
+which does not apply to the CVO. This can be accessed in different ways,
```
> This can be accessed in different ways, depending on the Operator.

This is just my observation so far from comparing the CLO and Samples Operator procedures. Need to verify if this is explaining things accurately.

sichvoge commented:
Wow there are actually more states than "managed" and "unmanaged". Not sure how many operators actually support all the other or even know that they exist.

Where did you get that info?

adellape commented:
> Where did you get that info?

@sichvoge The table for `managementState` was based on:

https://github.com/openshift/api/blob/11eee842dafd4698259c5de6b0746183ea6998f5/operator/v1/types.go#L27-L43

`Force` was new to me, but included in this first draft for completeness. Can drop that one if it's not worth it or wise to point out.

sichvoge commented:
I think the purpose for this part of the docs was to introduce an official statement re support when someone moves an operator to "unmanaged". I don't know if we want to explain all the possible states. 

Saying it differently, I am afraid that it would kinda bring the focus away from emphasizing the support statement to yet another API syntax documentation. Don't think we need that.

WDYT?

kalexand-rh commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+----
++
+The `spec.overrides[].unmanaged` parameter is set to `false` by default, however
+setting it to `true` blocks cluster upgrades and alerts the administrator that a
```
Does this alert display during update or when you set the override?

adellape replied:
Seems like it's a status condition during update (https://bugzilla.redhat.com/show_bug.cgi?id=1822844#c0) as I couldn't find it when I only set an override, but trying again to check.

adellape replied:
OK now I saw it this time, it just takes a few minutes. Confirmed that `Upgradeable=false` and the error message are added shortly after you set the override.

mburke5678 replied:
Should we show `unmanaged: false` in the code block as that is the default state? Presumably, if you are reading this for the first time and check  the CVO, it will say `false` (unless someone else changed it). 
Do we need a command to view/edit the CVO?

adellape replied:
@mburke5678 The whole `overrides` section actually isn't even there by default, the example snippet is just showing what it would look like it if it was added. I debated mentioning that too, might add.

lamek commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+While this can be helpful in non-production clusters or during debugging,
+Operators in an unmanaged state are unsupported and the cluster administrator
+assumes full control of the individual component configurations and upgrades.
+Depending on the method the Operator was set to unmanaged, it can also render
```
Some missing words here. Maybe:
"Depending on the method used to set the Operator to unmanaged, ..."

sferich888 replied:
I don't know that we want to say it 'render the entier cluster unsupported'. 
I get that this can be the case if the operator is a key system operator. However, this might not be the case in the majority of situations.

adellape replied:
@sferich888 OK, this was basically just alluding to if the method was by using a CVO override. I'll take out "Depending on the method the Operator was set to unmanaged, it can also render the entire cluster unsupported." and just leave the warning box later for "Setting CVO overrides puts the entire cluster in an unsupported state. Reported issues must be reproduced after removing the overrides for support to proceed," unless that also needs to be tweaked.

sferich888 commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+accomplishes this by modifying a Custom Resource (CR) that it manages, while the
+Samples Operator uses a cluster-wide configuration resource.
++
+The following management states are available with this method:
```
What are these based on (what are we documenting)? 
Are we documenting an example using the CR (like the Cluster Logging Operator), or the Samples Operator?

adellape replied:
https://github.com/openshift/openshift-docs/pull/21553#issuecomment-619930445

CLO and Samples Operator were just mentioned as examples, since I noticed the procedure for how they're set to unmanaged differed slightly.

sferich888 commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
++
+----
+Disabling ownership via cluster version overrides prevents upgrades. Please remove overrides before continuing.
+----
```
Do we need to document how to set CVO overides, the same way we show (above) how to set an 'unmanaged' state?

mburke5678 commented on modules/unmanaged-operators.adoc:
```diff
@@ -0,0 +1,96 @@
+unmet prerequisites exist. This state puts the Operator at risk for unsuccessful
+upgrades.
+|===
++
```
@adellape I am being told that the Cluster Logging Operator (and Elasticsearch operator, likely) does not support `Remove` and `Force` management states.

adellape replied:
OK thanks!

adellape commented:
Updated based on feedback: http://file.rdu.redhat.com/~adellape/042720/unmanaged_operators/architecture/architecture-installation.html#unmanaged-operators_architecture-installation

* Much more truncated
* Renamed section title to _Support policy for unmanaged Operators_.
* Flipped the bullets around so the `managementState` method is shown first
* Removed the module from the "Updating clusters" assemblies. Instead they're just an "Additional resources" link to the section in the Architecture guide. For example, the end of this section: http://file.rdu.redhat.com/~adellape/042720/unmanaged_operators/updating/updating-cluster-between-minor.html#update-service-overview_updating-cluster-between-minor 

Retained a copy of the longer version for comparison; some of this could be added back in some other form about all management states in the future: http://file.rdu.redhat.com/~adellape/042720-longer-version/unmanaged_operators/architecture/architecture-installation.html#unmanaged-operators_architecture-installation

sichvoge commented:
LGTM 👍

adellape commented:
/cherrypick enterprise-4.4

jiajliu commented:
The cvo part lgtm. About "Individual Operator configuration" part, thx for the confirm lgtm from @anpingli @xiuwang @xingxingxia

adellape commented:
/cherrypick enterprise-4.3

adellape commented:
/cherrypick enterprise-4.2
